lntest/itest/lnd_test.go

```diff
@@ -3826,12 +3826,19 @@ func testListPayments(net *lntest.NetworkHarness, t *harnessTest) {
 			p.PaymentHash, correctRHash)
 	}
 
-	// Finally, as we made a single-hop direct payment, there should have
-	// been no fee applied.
+	// As we made a single-hop direct payment, there should have been no fee
+	// applied.
 	if p.Fee != 0 {
 		t.Fatalf("incorrect Fee, got %v, want %v", p.Fee, 0)
 	}
 
+	// Finally, verify that the payment request returned by the rpc matches
+	// the invoice that we paid.
+	if p.PaymentRequest != invoiceResp.PaymentRequest {
```

Comment:
also add a check for a payment with no payment request?

Reply:
added